sys/vm/vm_map.c

```diff
@@ -2225,13 +2334,22 @@ vm_map_find(vm_map_t map, vm_object_t object, vm_ooffset_t offset,
 		alignment = (vm_offset_t)1 << (find_space >> 8);
 	} else
 		alignment = 0;
+
+#if __has_feature(capabilities)
+	if ((map->flags & MAP_RESERVATIONS) != 0) {
+		length = CHERI_REPRESENTABLE_LENGTH(length);
+		alignment = MAX(alignment,
+		    CHERI_REPRESENTABLE_ALIGNMENT(length));
+	}
+#endif
+
 	en_aslr = (map->flags & MAP_ASLR) != 0;
 	update_anon = cluster = clustering_anon_allowed(*addr) &&
 	    (map->flags & MAP_IS_SUB_MAP) == 0 && max_addr == 0 &&
 	    find_space != VMFS_NO_SPACE && object == NULL &&
 	    (cow & (MAP_INHERIT_SHARE | MAP_STACK_GROWS_UP |
 	    MAP_STACK_GROWS_DOWN)) == 0 && prot != PROT_NONE;
-	curr_min_addr = min_addr = *addr;
+	curr_min_addr = min_addr = vaddr = *addr;
```

Comment:
We could upstream the `vaddr` introduction; using `*addr` everywhere is a bit gross

Reply:
Split this in a separate commit to help with this.